daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -192,8 +219,50 @@ func (l *logStream) Close() error {
 	return nil
 }
 
-// create creates a log stream for the instance of the awslogs logging driver
+// create creates log group and log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+	if err := l.createLogStream(); err != nil {
+		if l.logCreateGroup {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == resourceNotFoundCode {
```

Comment:
@thaJeztah I would assume `resourceNotFoundCode` (which maps to "ResourceNotFoundException" in AWS error) covers MORE reasons than `log group does not exist`. (I would imagine resource refers more than just log group).

Under current implementation
1. Log stream creation, if succeed OK.
2. If log stream creation fails with `ResourceNotFoundException` (multiple scenarios), attempt to create log group.
3. If log group creation fails with `ResourceAlreadyExistsException`, this means log stream creation failure is not caused by `log groups does not exist`, retry log stream creation. 


The steps above will make 1-3 AWS calls depending on different scenarios.

Alternatively, we could:
1. Always check to see if log group exists with `func (*CloudWatchLogs) DescribeLogGroups`
2. If log group exist, then go ahead with log stream creation.
3. If log group does not exist, then go ahead with log group creation and log stream creation concurrently.

This approach will make 2-3 AWS calls depending on different scenarios.

It is hard to say which approach is better because they cover different cases. The overall average performance will depends on the ratio of different scenarios, and the cost of `DescribeLogGroups`.

Maybe @samuelkarp can provide some information about the likely performance preference?